src/TestSuite/IntegrationTestCase.php

```diff
@@ -424,8 +424,7 @@ protected function _buildRequest($url, $method, $data)
     protected function _addTokens($url, $data)
     {
         if ($this->_securityToken === true) {
-            $keys = Hash::flatten($data);
-            $tokenData = $this->_buildFieldToken($url, array_keys($keys));
+            $tokenData = $this->_buildFieldToken($url, array_keys($data));
```

Comment:
If using comments.0.author_id.
By not match the regular expression, do not become an unintended behavior?

At SecurityComponent.php _validatePost
 preg_match('/(\.\d+){1,10}$/', $key)